### PR TITLE
HIVE-24386 : Add builder methods for GetTablesRequest and GetPartitionsRequest to HiveMetaStoreClient

### PR DESCRIPTION
**Pull request by vnhive**
HIVE-24386 : Add builder methods for GetTablesRequest and GetPartitionsRequest to HiveMetaStoreClient

This patch builds over the patch for HIVE-24397 and adds builder methods for the request and the projection
specification classes of Tables and Partitions. The relevant unit tests have also been updated.

### CODE REVIEW

File: standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/TableFields.java [hunk 1]
```diff
@@ -58,9 +60,9 @@
 
     private static final ImmutableSet<String> allMultiValuedFields = new ImmutableSet.Builder<String>()
             .add("values")
-            .add("sd.cols.name")
-            .add("sd.cols.type")
-            .add("sd.cols.comment")
+            .add("sd.cd.cols.name")
```
Comment (nrg4878) [inline standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/TableFields.java:63]: @vnhive StorageDescriptor seems to have a variable "cols" but not the actual ColumnDescriptor "cd". How does this work?

File: standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/client/builder/GetPartitionsRequestBuilder.java [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+                                       String processorIdentifier, String validWriteIdList) {
+        this.catName = catName;
+        this.dbName = dbName;
+        this.tblName = tblName;
+        this.withAuth = withAuth;
+        this.user = user;
+        this.groupNames = groupNames;
+        this.projectionSpec = projectionSpec;
+        this.filterSpec = filterSpec;
+        this.processorCapabilities = processorCapabilities;
```
Comment (nrg4878) [inline standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/client/builder/GetPartitionsRequestBuilder.java:52]: the processorCapabilities and the processorIdentifier are meant to be static values for a given instance of HMSClient object. They are not meant to change between different calls. So having it on builder class might be confusing. Should we remove it?

File: standalone-metastore/metastore-server/src/test/java/org/apache/hadoop/hive/metastore/client/TestTablesGetExists.java [hunk 3]
```diff
@@ -555,6 +546,74 @@ public void testGetTableObjectsWithProjectionOfMultiValuedFields() throws Except
...
+      Assert.assertTrue(sd.isSetSerdeInfo());
+      Assert.assertTrue(sd.isSetBucketCols());
+      Assert.assertTrue(sd.isSetCompressed());
+      Assert.assertTrue(sd.isSetInputFormat());
+      Assert.assertTrue(sd.isSetSerdeInfo());
+      SerDeInfo serDeInfo = sd.getSerdeInfo();
+      Assert.assertTrue(serDeInfo.isSetSerializationLib());
+    }
+  }
+
```
Comment (nrg4878) [inline standalone-metastore/metastore-server/src/test/java/org/apache/hadoop/hive/metastore/client/TestTablesGetExists.java:616]: Could you please add a negative test here as well? Include only a couple of fields on the projection spec and test to ensure that some of the other fields are not set on the returned Table objects? Thanks

**Review by nrg4878 — CHANGES_REQUESTED** [review]
Added some comments. Requesting changes.

File: standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/TableFields.java (hunk 1 above)
Comment (vnhive) [reply standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/TableFields.java]: Sorry about the confusion naveen. I have reverted these back to the StorageDescriptor attributes. Thank you for the comment.

File: standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/client/builder/GetPartitionsRequestBuilder.java (hunk 2 above)
Comment (vnhive) [reply standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/client/builder/GetPartitionsRequestBuilder.java:52]: These seems harmless to me. I agree they are static, but then they are also public attributes that can be set. Having them in the builder methods is not very different from them being accessed through an object. I have retained them for now. However if you feel strongly about it, I will change it.

File: standalone-metastore/metastore-server/src/test/java/org/apache/hadoop/hive/metastore/client/TestTablesGetExists.java (hunk 3 above)
Comment (vnhive) [reply standalone-metastore/metastore-server/src/test/java/org/apache/hadoop/hive/metastore/client/TestTablesGetExists.java:643]: Done. I had changed all the test cases to use the builder methods these test negations also.

**Comment by vnhive** [comment]
> Added some comments. Requesting changes.

I have addressed all your requests. Can you please check if you are happy or you want me to change anything more ?

**Review by nrg4878 — APPROVED** [review]

**Review by pkumarsinha — APPROVED** [review]

**Pull request closed without merging**